### PR TITLE
ROX-15259: Do not show DNS flows in the inactive view

### PR DESCRIPTION
**Pull request by sachaudh**
## Description

This PR hides the DNS flows when we pick the `Inactive flows` option and we show it when we pick a new (third) option called `Inactive flows with DNS flows`

### CODE REVIEW

File: ui/apps/platform/src/Containers/NetworkGraph/NetworkGraphContainer.tsx [hunk 1]
```diff
@@ -222,6 +222,16 @@ function getDisplayEdges(edges: CustomEdgeModel[], showEdgeLabels: boolean): Cus
     });
 }
 
+function removeDNSFlows(edges: CustomEdgeModel[]) {
+    return edges.filter((edge) => {
+        return (
+            edge?.data?.sourceToTargetProperties[0]?.port !== 53 &&
+            edge?.data?.sourceToTargetProperties[0]?.port !== 5353 &&
+            edge?.data?.sourceToTargetProperties[0]?.protocol !== 'L4_PROTOCOL_UDP'
```
Comment (vjwilson) [inline ui/apps/platform/src/Containers/NetworkGraph/NetworkGraphContainer.tsx:230]: Is it possible that the `sourceToTargetProperties` array ever has more than one item, at index 0 ?

File: ui/apps/platform/src/Containers/NetworkGraph/NetworkGraphContainer.tsx [hunk 2]
```diff
@@ -313,12 +323,15 @@ function NetworkGraphContainer({
     let filteredNodes: CustomNodeModel[] = [...baseModel.nodes];
     let filteredEdges: CustomEdgeModel[] = [...baseModel.edges];
     // if edgeState is extraneous && there is a selectedNode, add in/egress flows nodes/edges
-    if (edgeState === 'inactive' && selectedNode?.data.type === 'DEPLOYMENT') {
+    if (edgeState !== 'active' && selectedNode?.data.type === 'DEPLOYMENT') {
```
Comment (vjwilson) [inline ui/apps/platform/src/Containers/NetworkGraph/NetworkGraphContainer.tsx:326]: Did you mean to flip the check here?

Before it had to be "inactive", but now it just has to be something other than "active" ?

File: ui/apps/platform/src/hooks/useFetchDeploymentCount.ts [hunk 3]
```diff
@@ -30,18 +31,10 @@ function useFetchDeploymentCount(selectedClusterId: string): UseFetchDeploymentC
...
-        }
-
-        setDeploymentCount(data.count);
-    }, [data]);
-
     return {
         loading,
         error,
-        deploymentCount,
+        deploymentCount: data?.cluster?.count || undefined,
```
Comment (vjwilson) [inline ui/apps/platform/src/hooks/useFetchDeploymentCount.ts:37]: Is the OK to be undefined, or does it need to default to 0 ?

**Review by vjwilson — COMMENTED** [review]
Some questions

File: ui/apps/platform/src/hooks/useFetchDeploymentCount.ts (hunk 3 above)
Comment (sachaudh) [reply ui/apps/platform/src/hooks/useFetchDeploymentCount.ts:37]: From the type it looked like it's either a `number` or `undefined` so I tried to keep it as expected. I believe it previously would be `undefined` until we set a number from the `useState<number>()`.

File: ui/apps/platform/src/Containers/NetworkGraph/NetworkGraphContainer.tsx (hunk 2 above)
Comment (sachaudh) [reply ui/apps/platform/src/Containers/NetworkGraph/NetworkGraphContainer.tsx:326]: Yep, I wanted to flip it. Before we had just `active` and `inactive`. So the `edgeState === 'inactive'` meant that it was `inactive`. Now that we technically have two `inactive` states, one with DNS and one without, the `edgeState !== 'active'` encompasses both those states. If we ever, God forbid, add more states, we'll have to redo this logic again 😢

File: ui/apps/platform/src/Containers/NetworkGraph/NetworkGraphContainer.tsx (hunk 1 above)
Comment (sachaudh) [reply ui/apps/platform/src/Containers/NetworkGraph/NetworkGraphContainer.tsx:230]: That's a good question. Now that I looked at this again, we also have a `targetToSourceProperties` field that only appears when it's a bidirectional edge. So I should double-check that there's no issue with that either. I'll look at this logic again

**Comment by sachaudh** [comment]
This looks like it may need more thought, so I'm going to close this PR for now

**Pull request closed without merging**